### PR TITLE
Increase maximum depth of outlining span search

### PR DESCRIPTION
**Pull request by uniqueiniquity**
This PR increases the maximum depth to which we search for outlining spans in the AST.
Averaged over 10 calls on tsserver.js on my machine, `getOutliningSpans` takes 388.873 ms per call with a maximum depth of 20 and 438.181 ms per call with a maximum depth of 120, resulting in a 12.68% slowdown. 120 was chosen because the maximum depth of the AST in tsserver.js is 107, and I believe our goal is for users to never have an outlining span fail to be created due to nesting depth. I'm certainly open to lowering it from 120 if desired.

### CODE REVIEW

File: tests/cases/fourslash/getOutliningSpans.ts [hunk 1]
```diff
@@ -89,12 +89,12 @@
...
 ////                                    module m10[| {
-////                                        module m11 {
-////                                            module m12 {
-////                                                export interface IFoo {
-////                                                }
-////                                            }
-////                                        }
+////                                        module m11[| {
+////                                            module m12[| {
+////                                                export interface IFoo[| {
```
Comment (billti) [inline tests/cases/fourslash/getOutliningSpans.ts]: Seems like there should still be a span that exceeds the limit which does NOT get outlined.

Comment (uniqueiniquity) [reply tests/cases/fourslash/getOutliningSpans.ts]: I agree, but I didn't want to get into another scenario with an unnecessarily large fourslash test. I think 120 would require 50 levels of nested modules to hit the limit. If we commit to setting the limit at 120, I'll go ahead and update the tests.

Comment (billti) [reply tests/cases/fourslash/getOutliningSpans.ts]: Why do they have to be nested modules? Can't you just have a nested array? e.g. keep going with something like:

```javascript
var x = [[[[[[[[[[[[[[[[[[
[
1,2,3
]
]]]]]]]]]]]]]]]]];
```

Comment (uniqueiniquity) [reply tests/cases/fourslash/getOutliningSpans.ts]: You're right, that should work fine.

**Pull request merged**